### PR TITLE
Add validation for some null arguments and dangling render passes

### PR DESCRIPTION
**Pull request by kainino0x**
And add tests for both

### CODE REVIEW

**Comment by kainino0x** [comment]
I'm pretty sure there are a bunch more null arguments checks that need to be added in many places, but this is a start.

File: src/tests/unittests/validation/CommandBufferValidationTests.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ValidationTest.h"
+
+class CommandBufferValidationTest : public ValidationTest {
+};
+
+TEST_F(CommandBufferValidationTest, Empty) {
```
Comment (Kangz) [inline src/tests/unittests/validation/CommandBufferValidationTests.cpp]: nit: comment before the tests (ANGLE-style).

**Comment by Kangz** [comment]
Null arguments are not allowed anywhere in the API currently and in fact the wire doesn't support them. I'll refactor this validation to but it in the autogenerated validation in templates/BackendProcTable.cpp.

Also I'm not sure how to split test file, for now I'm thinking of splitting them depending on which builder will generate the error, and maybe have several files for the command buffer builder if needed. What do you think?

**Comment by kainino0x** [comment]
null args: Okay, I was wondering about that. I've been having issues where if an invalid object is passed to a builder, it shows up on the validation side as a null pointer. I'll add comments and land, but you may want to remove these extra null checks once you add it to BackendProcTable.

split: Yes, I think that's the right split. That is (or was intended to be) how I split these files.

**Comment by Kangz** [comment]
Yep the split was fine, I was just clarifying for myself :)

**Pull request merged**